modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/TableDescriptorImpl.java

```diff
@@ -123,8 +124,23 @@ public TableDescriptorImpl(GridCacheContextInfo<?, ?> cacheInfo, GridQueryTypeDe
         // A _key/_val field is virtual in case there is an alias or a property(es) mapped to the _key/_val field.
         BitSet virtualFields = new BitSet();
 
-        descriptors.add(
-            new KeyValDescriptor(QueryUtils.KEY_FIELD_NAME, typeDesc.keyClass(), true, QueryUtils.KEY_COL));
+        if (typeDesc.implicitPk()) {
+            // pk is not set, thus we need to provide default value for autogenerated key
+            descriptors.add(
+                new KeyValDescriptor(QueryUtils.KEY_FIELD_NAME, typeDesc.keyClass(), true, QueryUtils.KEY_COL) {
+                    @Override public Object defaultValue() {
```

Comment:
possibly we need to overwrite this method too ? 
@Override public boolean hasDefaultValue() {
            return false;
        }